TorSharp/Tools/LineByLineConfigurer.cs

```diff
@@ -57,14 +57,24 @@ public LineByLineConfigurer(IConfigurationDictionary configurationDictionary, IC
                     }
                 }
 
+                // If the original file exists, remove it before copying over the temporary file.
                 if (File.Exists(path))
                 {
-                    File.Replace(temporaryPath, path, null);
-                }
-                else
-                {
-                    File.Move(temporaryPath, path);
+                    string backupPath = path + ".bak";
+
+                    // If there has already been a backup, just delete the original.
+                    if (File.Exists(backupPath))
+                    {
+                        File.Delete(path);
```

Comment:
Going for a solution that specified `File.Replace(temporaryPath, path, backupPath);` was my first attempt, but it didn't work out. I never managed to work out exactly why it was failing, as I said, through trial and error this version worked.

I then went for the option of the backup being of the original (if there is one) config file. The rational being: if it ever breaks, having the original config file as a comparison is useful, though granted it is available in the downloaded zip.

I'm happy to change the backup strategy to being a backup of the previous execution rather than a back of the original, but I wasn't able to get `File.Replace` working in any way.

Let me know know, and I'll amend the back to be:
```c#
// If there has already been a backup, delete it.
if (File.Exists(backupPath))
{
    File.Delete(backupPath);
}
// Backup the last config.
File.Move(path, backupPath);
```